<!DOCTYPE html><html lang="zh-CN" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no"><title>4.2 Sequence Reply and Mutation Identification | 梁止潆的博客</title><meta name="keywords" content="BioInformatics,Learning Note"><meta name="author" content="Joyce"><meta name="copyright" content="Joyce"><meta name="format-detection" content="telephone=no"><meta name="theme-color" content="#ffffff"><meta name="description" content="4.2 序列回帖和变异鉴定 NGS：Reads Mapping   Reads Mapping 是啥？ 是指测序得到的DNA片段也就是Reads，定位到基因组上，通过Reads Mapping 在克服了深度测序产生的Reads过短导致的技术困难的同时，也方便利用基因组位置作为桥梁来将测序得到的数据与前期研究产生的注释结果进行有机的整合   作用 往往被作为深度测序数据分析的第一步，其质量的好坏以及">
<meta property="og:type" content="article">
<meta property="og:title" content="4.2 Sequence Reply and Mutation Identification">
<meta property="og:url" content="http://joyceliang.club/2022/4-2-Sequence-Reply-and-Mutation-Identification/index.html">
<meta property="og:site_name" content="梁止潆的博客">
<meta property="og:description" content="4.2 序列回帖和变异鉴定 NGS：Reads Mapping   Reads Mapping 是啥？ 是指测序得到的DNA片段也就是Reads，定位到基因组上，通过Reads Mapping 在克服了深度测序产生的Reads过短导致的技术困难的同时，也方便利用基因组位置作为桥梁来将测序得到的数据与前期研究产生的注释结果进行有机的整合   作用 往往被作为深度测序数据分析的第一步，其质量的好坏以及">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="http://img.peterli.club/joy/202210191632673.png">
<meta property="article:published_time" content="2022-10-19T08:36:34.000Z">
<meta property="article:modified_time" content="2022-10-19T08:45:06.722Z">
<meta property="article:author" content="Joyce">
<meta property="article:tag" content="BioInformatics">
<meta property="article:tag" content="Learning Note">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="http://img.peterli.club/joy/202210191632673.png"><link rel="shortcut icon" href="http://img.peterli.club/joy/202210141514731.jpeg"><link rel="canonical" href="http://joyceliang.club/2022/4-2-Sequence-Reply-and-Mutation-Identification/"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//hm.baidu.com"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><meta name="baidu-site-verification" content="code-c9hz390Wld"/><link rel="stylesheet" href="/css/index.css"><link rel="stylesheet" href="https://unpkg.com/@fortawesome/fontawesome-free/css/all.min.css" media="print" onload="this.media='all'"><link rel="stylesheet" href="https://unpkg.com/@fancyapps/ui/dist/fancybox.css" media="print" onload="this.media='all'"><script>var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "https://hm.baidu.com/hm.js?e50dee5804918429b74a301c425729fe";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script><script>const GLOBAL_CONFIG = { 
  root: '/',
  algolia: undefined,
  localSearch: {"path":"/search.json","preload":false,"languages":{"hits_empty":"找不到您查询的内容：${query}"}},
  translate: undefined,
  noticeOutdate: undefined,
  highlight: {"plugin":"highlighjs","highlightCopy":true,"highlightLang":true,"highlightHeightLimit":false},
  copy: {
    success: '复制成功',
    error: '复制错误',
    noSupport: '浏览器不支持'
  },
  relativeDate: {
    homepage: false,
    post: false
  },
  runtime: '天',
  date_suffix: {
    just: '刚刚',
    min: '分钟前',
    hour: '小时前',
    day: '天前',
    month: '个月前'
  },
  copyright: undefined,
  lightbox: 'fancybox',
  Snackbar: undefined,
  source: {
    justifiedGallery: {
      js: 'https://unpkg.com/flickr-justified-gallery/dist/fjGallery.min.js',
      css: 'https://unpkg.com/flickr-justified-gallery/dist/fjGallery.css'
    }
  },
  isPhotoFigcaption: false,
  islazyload: false,
  isAnchor: false
}</script><script id="config-diff">var GLOBAL_CONFIG_SITE = {
  title: '4.2 Sequence Reply and Mutation Identification',
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isToc: true,
  postUpdate: '2022-10-19 16:45:06'
}</script><noscript><style type="text/css">
  #nav {
    opacity: 1
  }
  .justified-gallery img {
    opacity: 1
  }

  #recent-posts time,
  #post-meta time {
    display: inline !important
  }
</style></noscript><script>(win=>{
    win.saveToLocal = {
      set: function setWithExpiry(key, value, ttl) {
        if (ttl === 0) return
        const now = new Date()
        const expiryDay = ttl * 86400000
        const item = {
          value: value,
          expiry: now.getTime() + expiryDay,
        }
        localStorage.setItem(key, JSON.stringify(item))
      },

      get: function getWithExpiry(key) {
        const itemStr = localStorage.getItem(key)

        if (!itemStr) {
          return undefined
        }
        const item = JSON.parse(itemStr)
        const now = new Date()

        if (now.getTime() > item.expiry) {
          localStorage.removeItem(key)
          return undefined
        }
        return item.value
      }
    }
  
    win.getScript = url => new Promise((resolve, reject) => {
      const script = document.createElement('script')
      script.src = url
      script.async = true
      script.onerror = reject
      script.onload = script.onreadystatechange = function() {
        const loadState = this.readyState
        if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
        script.onload = script.onreadystatechange = null
        resolve()
      }
      document.head.appendChild(script)
    })
  
      win.activateDarkMode = function () {
        document.documentElement.setAttribute('data-theme', 'dark')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#0d0d0d')
        }
      }
      win.activateLightMode = function () {
        document.documentElement.setAttribute('data-theme', 'light')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#ffffff')
        }
      }
      const t = saveToLocal.get('theme')
    
          if (t === 'dark') activateDarkMode()
          else if (t === 'light') activateLightMode()
        
      const asideStatus = saveToLocal.get('aside-status')
      if (asideStatus !== undefined) {
        if (asideStatus === 'hide') {
          document.documentElement.classList.add('hide-aside')
        } else {
          document.documentElement.classList.remove('hide-aside')
        }
      }
    
    const detectApple = () => {
      if(/iPad|iPhone|iPod|Macintosh/.test(navigator.userAgent)){
        document.documentElement.classList.add('apple')
      }
    }
    detectApple()
    })(window)</script><meta name="generator" content="Hexo 6.3.0"></head><body><div id="sidebar"><div id="menu-mask"></div><div id="sidebar-menus"><div class="avatar-img is-center"><img src="http://img.peterli.club/joy/202210141514731.jpeg" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="sidebar-site-data site-data is-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">43</div></a><a href="/tags/"><div class="headline">标签</div><div class="length-num">8</div></a><a href="/categories/"><div class="headline">分类</div><div class="length-num">10</div></a></div><hr/><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> 主页</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fas fa-archive"></i><span> 时间轴</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fas fa-tags"></i><span> 标签</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page group" href="javascript:void(0);"><i class="fa-fw fas fa-list"></i><span> 目录</span><i class="fas fa-chevron-down"></i></a><ul class="menus_item_child"><li><a class="site-page child" href="/paper/"><i class="fa-fw fas fa-duotone fa-newspaper"></i><span> Paper</span></a></li><li><a class="site-page child" href="/bioinformatics/"><i class="fa-fw fas fa-regular fa-dna"></i><span> BioInformatics</span></a></li></ul></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-heart"></i><span> 关于</span></a></div></div></div></div><div class="post" id="body-wrap"><header class="post-bg" id="page-header" style="background-image: url('http://img.peterli.club/joy/202210191632673.png')"><nav id="nav"><span id="blog_name"><a id="site-name" href="/">梁止潆的博客</a></span><div id="menus"><div id="search-button"><a class="site-page social-icon search"><i class="fas fa-search fa-fw"></i><span> 搜索</span></a></div><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> 主页</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fas fa-archive"></i><span> 时间轴</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fas fa-tags"></i><span> 标签</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page group" href="javascript:void(0);"><i class="fa-fw fas fa-list"></i><span> 目录</span><i class="fas fa-chevron-down"></i></a><ul class="menus_item_child"><li><a class="site-page child" href="/paper/"><i class="fa-fw fas fa-duotone fa-newspaper"></i><span> Paper</span></a></li><li><a class="site-page child" href="/bioinformatics/"><i class="fa-fw fas fa-regular fa-dna"></i><span> BioInformatics</span></a></li></ul></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-heart"></i><span> 关于</span></a></div></div><div id="toggle-menu"><a class="site-page"><i class="fas fa-bars fa-fw"></i></a></div></div></nav><div id="post-info"><h1 class="post-title">4.2 Sequence Reply and Mutation Identification</h1><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">发表于</span><time class="post-meta-date-created" datetime="2022-10-19T08:36:34.000Z" title="发表于 2022-10-19 16:36:34">2022-10-19</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">更新于</span><time class="post-meta-date-updated" datetime="2022-10-19T08:45:06.722Z" title="更新于 2022-10-19 16:45:06">2022-10-19</time></span><span class="post-meta-categories"><span class="post-meta-separator">|</span><i class="fas fa-inbox fa-fw post-meta-icon"></i><a class="post-meta-categories" href="/categories/Learning-Note/">Learning Note</a><i class="fas fa-angle-right post-meta-separator"></i><i class="fas fa-inbox fa-fw post-meta-icon"></i><a class="post-meta-categories" href="/categories/Learning-Note/BioInformatics/">BioInformatics</a><i class="fas fa-angle-right post-meta-separator"></i><i class="fas fa-inbox fa-fw post-meta-icon"></i><a class="post-meta-categories" href="/categories/Learning-Note/BioInformatics/PKUBioInformatics/">PKUBioInformatics</a></span></div><div class="meta-secondline"><span class="post-meta-separator">|</span><span class="post-meta-pv-cv" id="" data-flag-title="4.2 Sequence Reply and Mutation Identification"><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">阅读量:</span><span id="busuanzi_value_page_pv"><i class="fa-solid fa-spinner fa-spin"></i></span></span></div></div></div></header><main class="layout" id="content-inner"><div id="post"><article class="post-content" id="article-container"><h1>4.2 序列回帖和变异鉴定 NGS：Reads Mapping</h1>
<ul>
<li>
<p><strong>Reads Mapping 是啥？</strong></p>
<p>是指测序得到的DNA片段也就是Reads，定位到基因组上，通过Reads Mapping 在克服了深度测序产生的Reads过短导致的技术困难的同时，也方便利用基因组位置作为桥梁来将测序得到的数据与前期研究产生的注释结果进行有机的整合</p>
</li>
<li>
<p><strong>作用</strong></p>
<p>往往被作为深度测序数据分析的第一步，其质量的好坏以及速度的快慢都会对后续的分析产生影响</p>
<img src="http://img.peterli.club/joy/202210191637467.png" alt="Untitled" style="zoom: 33%;" />
</li>
<li>
<p><strong>Mapping:</strong> <strong>Input Data</strong></p>
<p>本质上还是双序列比对问题</p>
<ul>
<li>
<p>Reference Genome</p>
<p>— Nucleotide</p>
<p>— <strong>Length</strong>: Hundreds of Mb per chromosome</p>
<p>— ～3Gb in total(for human genome)</p>
</li>
<li>
<p>Reads</p>
<p>— Nucleotide, with <strong>various qualities</strong>(relatively <strong>high error rate</strong> 1e-2 ~ 1e-5)</p>
<p>— <strong>Length</strong>: 36~80 bp per read</p>
<p>— Hundreds of Gbs per run</p>
</li>
</ul>
<img src="http://img.peterli.club/joy/202210191637461.png" alt="Untitled" style="zoom: 25%;" />
<img src="http://img.peterli.club/joy/202210191637461.png" alt="Untitled" style="zoom:25%;" />
</li>
<li>
<p>BLAST Ideas: Seeding-and-extending</p>
<p>通过对基因组进行索引，从而将Reads快速定位，而后再通过标准动态规划算法来构建最终的alignment</p>
<ol>
<li>
<p>Find matches(seed) between the query and subject 寻找查询和主题之间的匹配</p>
</li>
<li>
<p>Extend seed into High Scoring Segment Pairs(HSPs) 将种子扩展为高分段对(HSP)</p>
<p>— Run Smith-Waterman algorithm on the specified region only 仅在指定区域内运行Smith-Waterman算法</p>
</li>
<li>
<p>Assess the reliability of the alignment 评估对齐的可靠性</p>
</li>
</ol>
<img src="http://img.peterli.club/joy/202210191637462.png" alt="Untitled" style="zoom:33%;" />
<ul>
<li>
<p>数据索引</p>
<p>所谓的索引本质上就是对数据的分组，基于对原始数据中的keys应用索引函数来进行处理，可以将原始数据划分为若干更小的组，并通过将与keys相关的检索限制在某个小组里，来缩小搜索空间，从而降低搜索时间</p>
<img src="http://img.peterli.club/joy/202210191637466.png" alt="Untitled" style="zoom: 25%;" />
<ul>
<li>
<p>哈希</p>
<img src="http://img.peterli.club/joy/202210191637464.png" alt="Untitled" style="zoom: 25%;" />
<p>例：对一段基因组序列，首先参考BLAST思路，将之切分成若干的细的片段，而后针对每一个细的片段计算其哈希值作为其在索引表中的地址，并在地址表中保存这段序列及其在基因组中的坐标，在Reads中再次出现这个片段的时候就可以在地址表中迅速找到其在基因组中出现的位置</p>
<img src="http://img.peterli.club/joy/202210191637755.png" alt="Untitled" style="zoom: 25%;" />
</li>
<li>
<p>抽屉原理 提高Reads Mapping速度</p>
<p>在Reads Mapping过程中，为了提高灵敏度，通常我们会允许若干碱基的错配，这时我们就可以依据抽屉原理，将Reads划分为若干不重叠的块。</p>
<img src="http://img.peterli.club/joy/202210191637784.png" alt="Untitled" style="zoom: 25%;" />
<p>例如我们将Reads划分为3个不重叠的块，那么就可以确保在错配不超过两个的情况下，至少有一个片段可以作为与基因组完全匹配的种子来查询索引，从而大大提高了Reads Mapping的速度</p>
<img src="http://img.peterli.club/joy/202210191637834.png" alt="Untitled" style="zoom:25%;" />
</li>
<li>
<p>前缀树、后缀树 —— 大大提高内存利用的效率和比对的速度</p>
<p>早期的Reads  Mapping算法ELAND、MAQ、SOAP1等等均广泛的应用了这个思路，但是如果搜索的时候允许较多的错配位点，就需要把序列分成很多小的区块，那这会导致性能急剧下降。因此，09年开始用于数据压缩算法的前缀树和后缀树开始被应用于Reads Mapping，这些数据结构通过合并共享子串降低内存消耗并且可以方便实现最长公共子串的查找。</p>
<img src="http://img.peterli.club/joy/202210191637869.png" alt="Untitled" style="zoom:25%;" />
<p>例BOWTIE、BWA、SOAP3等工具均采用了基于后缀的BWT转换，可以逐位对比对片段基因延伸，大大提高内存利用的效率和比对的速度</p>
<img src="http://img.peterli.club/joy/202210191637891.png" alt="Untitled" style="zoom:25%;" />
</li>
</ul>
</li>
<li>
<p>找位置</p>
<p>通过索引找到相应的地址之后，我们就可以参照之前BLAST思路，向左右两个方向延伸扩展之后用动态规划算法来确定最后的比对</p>
<img src="http://img.peterli.club/joy/202210191637066.png" alt="Untitled" style="zoom: 25%;" />
<p>与BLAST搜索不同的是，新一代测序技术的错误率很高，对于一个测序位点我们就需要考虑它是由测序错误引起的假象的可能性，因此08年提出了Mapping Quality 分数的概念来处理这一问题</p>
<img src="http://img.peterli.club/joy/202210191637089.png" alt="Untitled" style="zoom:25%;" />
<p>首先假定看到的错配都是由于测序错误引起的，这个时候的概率就是各个错配碱基所对应的概率乘积，将其做过log transformation后的数值称之为SQ。</p>
<img src="http://img.peterli.club/joy/202210191637228.png" alt="Untitled" style="zoom:25%;" />
<p>在所有的Mapping位置都计算了一个相应的SQ值之后，就可以进一步的对每个位置计算出错误mapping的概率E，这里计算E的时候是同时考虑序列的相似程度和测序质量，因此在实际的测序中，会更多的使用Mapping Quality而不是序列比对分数来筛选真正的Reads Mapping的位置</p>
<img src="http://img.peterli.club/joy/202210191637250.png" alt="Untitled" style="zoom:25%;" />
</li>
</ul>
</li>
<li>
<p>Genetic Variants —— 遗传变异</p>
<p>当将测序得到的Reads正确Map到基因组之后，就可以开始来鉴定遗传变异，根据遗传变异的尺寸，可以将之分为在单个碱基水平的单核苷酸变异即SNV，和涉及到多个碱基的结构变异SV</p>
<ul>
<li>
<p>SNV：Single Nucleotide Variant  单核苷酸变体</p>
<p>— Substitution(SNP)  碱基替换</p>
<p>— Indel: insertion/deletion</p>
</li>
<li>
<p>Structural Variation(SV)  结构变异</p>
<p>— Large-scale insertion/deletion  大规模插入/删除</p>
<p>— Inversion  倒置</p>
<p>— Translocation  易位</p>
<p>— Copy Number Variation(CNV)  拷贝数变异</p>
</li>
</ul>
<img src="http://img.peterli.club/joy/202210191637279.png" alt="Untitled" style="zoom: 50%;" />
<ul>
<li>
<p>SNP Calling is NOT Genotyping</p>
<ul>
<li>
<p>&quot;SNP calling aims to determine in which positions there are polymorphisms or in which positions at least one of the bases differs from a reference sequence”</p>
<p>&quot;SNP调用的目的是确定在哪些位置存在多态性，或在哪些位置至少有一个碱基与参考序列不同”</p>
</li>
<li>
<p>&quot;Genotype calling is the process of determining the genotype for each individual and is typically only done for positions in which a SNP or a ‘variant’ has already been called.”</p>
<p>&quot;基因型调用是确定每个个体的基因型的过程，通常只针对已经调用了SNP或’变体’的位置。”</p>
</li>
</ul>
</li>
<li>
<p>Counting: an intuitive (and naïve) approach</p>
<p>在Reads被正确Map的前提下，我们可以参考特定位点上的Map的Reads，运用经验规则来基于计数的方法同时完成SNP calling和Genotyping</p>
<img src="http://img.peterli.club/joy/202210191637305.png" alt="Untitled" style="zoom: 33%;" />
<p>例：</p>
<ul>
<li>
<p>Counting high-confident, non - reference allele (i.e. Quality &gt;= 20)</p>
<p>—  Freq(突变位点的频率) &lt;20% or &gt; 80%: homozygous genotype 纯合子变异</p>
<p>— Otherwise: heterozygous 杂合变异位点</p>
</li>
<li>
<p>Works well for “deeply sequenced regions” (DSR), i.e. depth &gt; 25x</p>
<p>对 “深度测序区域”（DSR）效果良好，即深度大于25倍</p>
<p>— But suffer from under-calling of heterozygous genotypes for low -coverage regions</p>
<p>但对低覆盖率区域的杂合基因型则存在低估的问题</p>
<p>— And can’t give an objective measurement for reliability</p>
<p>不能对可靠性进行客观的测量</p>
</li>
</ul>
</li>
<li>
<p>A Simple Probabilistic Model for Genotyping</p>
<ol>
<li>
<p>For a diploid genome, there will be at most two different alleles (A and a)observed at a given site:</p>
<p>对于一个二倍体基因组，在一个给定的部位最多可以观察到两种不同的等位基因(A和a)</p>
<p>— 3 possible genotypes: &lt;A,A&gt;, &lt;A,a&gt;, &lt;a,a&gt; — Number of A： k;  Number of a:  n-k</p>
</li>
<li>
<p>Then, the probability for each genotypes is</p>
<img src="http://img.peterli.club/joy/202210191637523.png" alt="Untitled" style="zoom:33%;" />
</li>
<li>
<p>Bayes Formula can be further employed to calculate posterior probabilities, i.e. P(&lt;A,A&gt;| D), P(&lt;a, a&gt;|D), and P(&lt;A,a&gt;| D) if we can estimate the prior probabilities P(&lt;A,A&gt;), P(&lt;a,a&gt;) and P(&lt;A,a&gt;)</p>
</li>
</ol>
</li>
</ul>
<img src="http://img.peterli.club/joy/202210191637673.png" alt="Untitled" style="zoom: 25%;" />
</li>
</ul>
</article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">文章作者: </span><span class="post-copyright-info"><a href="http://joyceliang.club">Joyce</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">文章链接: </span><span class="post-copyright-info"><a href="http://joyceliang.club/2022/4-2-Sequence-Reply-and-Mutation-Identification/">http://joyceliang.club/2022/4-2-Sequence-Reply-and-Mutation-Identification/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">版权声明: </span><span class="post-copyright-info">本博客所有文章除特别声明外，均采用 <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank">CC BY-NC-SA 4.0</a> 许可协议。转载请注明来自 <a href="http://joyceliang.club" target="_blank">梁止潆的博客</a>！</span></div></div><div class="tag_share"><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/BioInformatics/">BioInformatics</a><a class="post-meta__tags" href="/tags/Learning-Note/">Learning Note</a></div><div class="post_share"><div class="social-share" data-image="http://img.peterli.club/joy/202210191632673.png" data-sites="facebook,twitter,wechat,weibo,qq"></div><link rel="stylesheet" href="https://unpkg.com/butterfly-extsrc/sharejs/dist/css/share.min.css" media="print" onload="this.media='all'"><script src="https://unpkg.com/butterfly-extsrc/sharejs/dist/js/social-share.min.js" defer></script></div></div><nav class="pagination-post" id="pagination"><div class="prev-post pull-left"><a href="/2022/4-3-Analysis-and-Demonstration/"><img class="prev-cover" src="http://img.peterli.club/joy/202210191632673.png" onerror="onerror=null;src='/img/404.jpg'" alt="cover of previous post"><div class="pagination-info"><div class="label">上一篇</div><div class="prev_info">4.3 Analysis and Demonstration</div></div></a></div><div class="next-post pull-right"><a href="/2022/4-1-Next-Generation-Sequencing/"><img class="next-cover" src="http://img.peterli.club/joy/202210191632673.png" onerror="onerror=null;src='/img/404.jpg'" alt="cover of next post"><div class="pagination-info"><div class="label">下一篇</div><div class="next_info">4.1 Next Generation Sequencing</div></div></a></div></nav><div class="relatedPosts"><div class="headline"><i class="fas fa-thumbs-up fa-fw"></i><span>相关推荐</span></div><div class="relatedPosts-list"><div><a href="/2022/1-3-DNA-Replication/" title="1.3 DNA Replication"><img class="cover" src="http://img.peterli.club/joy/202210201030299.png" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2022-10-20</div><div class="title">1.3 DNA Replication</div></div></a></div><div><a href="/2022/1-4-The-Chemistry-of-DNA-Replication/" title="1.4 The Chemistry of DNA Replication"><img class="cover" src="http://img.peterli.club/joy/202210201301024.png" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2022-10-20</div><div class="title">1.4 The Chemistry of DNA Replication</div></div></a></div><div><a href="/2022/1-2-DNA-Structure/" title="1.2 DNA Structure"><img class="cover" src="http://img.peterli.club/joy/202210201030299.png" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2022-10-20</div><div class="title">1.2 DNA Structure</div></div></a></div><div><a href="/2022/1-5-The-DNA-Polymerase-Active-Site/" title="1.5 The DNA Polymerase Active Site"><img class="cover" src="http://img.peterli.club/joy/202210201319055.png" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2022-10-20</div><div class="title">1.5 The DNA Polymerase Active Site</div></div></a></div><div><a href="/2022/1-6and7-The-Structure-of-DNA-Polymerase/" title="1.6&amp;7 The Structure of DNA Polymerase"><img class="cover" src="http://img.peterli.club/joy/202210201347160.png" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2022-10-20</div><div class="title">1.6&amp;7 The Structure of DNA Polymerase</div></div></a></div><div><a href="/2022/1-8and9-Mistakes-of-DNA-Polymerase/" title="1.8and9 Mistakes of DNA Polymerase"><img class="cover" src="http://img.peterli.club/joy/202210201403607.png" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2022-10-20</div><div class="title">1.8and9 Mistakes of DNA Polymerase</div></div></a></div></div></div><hr/><div id="post-comment"><div class="comment-head"><div class="comment-headline"><i class="fas fa-comments fa-fw"></i><span> 评论</span></div></div><div class="comment-wrap"><div><div id="gitalk-container"></div></div></div></div></div><div class="aside-content" id="aside-content"><div class="card-widget card-info"><div class="is-center"><div class="avatar-img"><img src="http://img.peterli.club/joy/202210141514731.jpeg" onerror="this.onerror=null;this.src='/img/friend_404.gif'" alt="avatar"/></div><div class="author-info__name">Joyce</div><div class="author-info__description">上海科技大学 | 生物医学工程学院 | 生物信息学</div></div><div class="card-info-data site-data is-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">43</div></a><a href="/tags/"><div class="headline">标签</div><div class="length-num">8</div></a><a href="/categories/"><div class="headline">分类</div><div class="length-num">10</div></a></div><a id="card-info-btn" target="_blank" rel="noopener" href="https://github.com/JoyceLiang-sudo"><i class="fab fa-github"></i><span>Follow Me</span></a><div class="card-info-social-icons is-center"><a class="social-icon" href="https://github.com/JoyceLiang-sudo" target="_blank" title="Github"><i class="fab fa-github"></i></a><a class="social-icon" href="mailto:joy.sudo996@gmail.com" target="_blank" title="Email"><i class="fas fa-envelope"></i></a></div></div><div class="card-widget card-announcement"><div class="item-headline"><i class="fas fa-bullhorn fa-shake"></i><span>公告</span></div><div class="announcement_content">不会敲代码也不会分子生物学的生信狗</div></div><div class="sticky_layout"><div class="card-widget" id="card-toc"><div class="item-headline"><i class="fas fa-stream"></i><span>目录</span><span class="toc-percentage"></span></div><div class="toc-content"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link"><span class="toc-number">1.</span> <span class="toc-text">4.2 序列回帖和变异鉴定 NGS：Reads Mapping</span></a></li></ol></div></div><div class="card-widget card-recent-post"><div class="item-headline"><i class="fas fa-history"></i><span>最新文章</span></div><div class="aside-list"><div class="aside-list-item"><a class="thumbnail" href="/2022/2-10-Dedicated-database-KEGG-OMIM/" title="2.10 Dedicated database KEGG, OMIM"><img src="http://img.peterli.club/joy/202210211052313.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="2.10 Dedicated database KEGG, OMIM"/></a><div class="content"><a class="title" href="/2022/2-10-Dedicated-database-KEGG-OMIM/" title="2.10 Dedicated database KEGG, OMIM">2.10 Dedicated database KEGG, OMIM</a><time datetime="2022-10-21T02:51:47.000Z" title="发表于 2022-10-21 10:51:47">2022-10-21</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2022/2-9-Secondary-Protein-Database/" title="2.9 Secondary Protein Database"><img src="http://img.peterli.club/joy/202210211034685.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="2.9 Secondary Protein Database"/></a><div class="content"><a class="title" href="/2022/2-9-Secondary-Protein-Database/" title="2.9 Secondary Protein Database">2.9 Secondary Protein Database</a><time datetime="2022-10-21T02:32:39.000Z" title="发表于 2022-10-21 10:32:39">2022-10-21</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2022/2-8-Primary-Protein-Structure-Database-PDB/" title="2.8 Primary Protein Structure Database PDB"><img src="http://img.peterli.club/joy/202210211024487.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="2.8 Primary Protein Structure Database PDB"/></a><div class="content"><a class="title" href="/2022/2-8-Primary-Protein-Structure-Database-PDB/" title="2.8 Primary Protein Structure Database PDB">2.8 Primary Protein Structure Database PDB</a><time datetime="2022-10-21T02:22:30.000Z" title="发表于 2022-10-21 10:22:30">2022-10-21</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2022/2-7-Primary-Protein-Sequence-Database-UniProt/" title="2.7 Primary Protein Sequence Database UniProt"><img src="http://img.peterli.club/joy/202210211008928.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="2.7 Primary Protein Sequence Database UniProt"/></a><div class="content"><a class="title" href="/2022/2-7-Primary-Protein-Sequence-Database-UniProt/" title="2.7 Primary Protein Sequence Database UniProt">2.7 Primary Protein Sequence Database UniProt</a><time datetime="2022-10-21T02:07:31.000Z" title="发表于 2022-10-21 10:07:31">2022-10-21</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2022/2-6-Secondary-Nucleic-Acid-Database/" title="2.6 Secondary Nucleic Acid Database"><img src="http://img.peterli.club/joy/202210210953987.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="2.6 Secondary Nucleic Acid Database"/></a><div class="content"><a class="title" href="/2022/2-6-Secondary-Nucleic-Acid-Database/" title="2.6 Secondary Nucleic Acid Database">2.6 Secondary Nucleic Acid Database</a><time datetime="2022-10-21T01:53:02.000Z" title="发表于 2022-10-21 09:53:02">2022-10-21</time></div></div></div></div></div></div></main><footer id="footer"><div id="footer-wrap"><div class="copyright">&copy;2019 - 2022 By Joyce</div><div class="framework-info"><span>框架 </span><a target="_blank" rel="noopener" href="https://hexo.io">Hexo</a><span class="footer-separator">|</span><span>主题 </span><a target="_blank" rel="noopener" href="https://github.com/jerryc127/hexo-theme-butterfly">Butterfly</a></div><div class="footer_custom_text"><a target="_blank" rel="noopener" href="http://www.beian.miit.gov.cn/"><img class="icp-icon" src="http://www.beian.gov.cn/img/new/gongan.png"><span>鲁ICP备19056758号-1</span></a></div></div></footer></div><div id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="阅读模式"><i class="fas fa-book-open"></i></button><button id="darkmode" type="button" title="浅色和深色模式转换"><i class="fas fa-adjust"></i></button><button id="hide-aside-btn" type="button" title="单栏和双栏切换"><i class="fas fa-arrows-alt-h"></i></button></div><div id="rightside-config-show"><button id="rightside_config" type="button" title="设置"><i class="fas fa-cog fa-spin"></i></button><button class="close" id="mobile-toc-button" type="button" title="目录"><i class="fas fa-list-ul"></i></button><a id="to_comment" href="#post-comment" title="直达评论"><i class="fas fa-comments"></i></a><button id="go-up" type="button" title="回到顶部"><i class="fas fa-arrow-up"></i></button></div></div><div id="local-search"><div class="search-dialog"><nav class="search-nav"><span class="search-dialog-title">搜索</span><span id="loading-status"></span><button class="search-close-button"><i class="fas fa-times"></i></button></nav><div class="is-center" id="loading-database"><i class="fas fa-spinner fa-pulse"></i><span>  数据库加载中</span></div><div class="search-wrap"><div id="local-search-input"><div class="local-search-box"><input class="local-search-box--input" placeholder="搜索文章" type="text"/></div></div><hr/><div id="local-search-results"></div></div></div><div id="search-mask"></div></div><div><script src="/js/utils.js"></script><script src="/js/main.js"></script><script src="https://unpkg.com/@fancyapps/ui/dist/fancybox.umd.js"></script><script src="/js/search/local-search.js"></script><div class="js-pjax"><script>if (!window.MathJax) {
  window.MathJax = {
    tex: {
      inlineMath: [ ['$','$'], ["\\(","\\)"]],
      tags: 'ams'
    },
    chtml: {
      scale: 1.2
    },
    options: {
      renderActions: {
        findScript: [10, doc => {
          for (const node of document.querySelectorAll('script[type^="math/tex"]')) {
            const display = !!node.type.match(/; *mode=display/)
            const math = new doc.options.MathItem(node.textContent, doc.inputJax[0], display)
            const text = document.createTextNode('')
            node.parentNode.replaceChild(text, node)
            math.start = {node: text, delim: '', n: 0}
            math.end = {node: text, delim: '', n: 0}
            doc.math.push(math)
          }
        }, ''],
        insertScript: [200, () => {
          document.querySelectorAll('mjx-container:not\([display]\)').forEach(node => {
            const target = node.parentNode
            if (target.nodeName.toLowerCase() === 'li') {
              target.parentNode.classList.add('has-jax')
            } else {
              target.classList.add('has-jax')
            }
          });
        }, '', false]
      }
    }
  }
  
  const script = document.createElement('script')
  script.src = 'https://unpkg.com/mathjax/es5/tex-mml-chtml.js'
  script.id = 'MathJax-script'
  script.async = true
  document.head.appendChild(script)
} else {
  MathJax.startup.document.state(0)
  MathJax.texReset()
  MathJax.typeset()
}</script><script>function addGitalkSource () {
  const ele = document.createElement('link')
  ele.rel = 'stylesheet'
  ele.href= 'https://unpkg.com/gitalk/dist/gitalk.css'
  document.getElementsByTagName('head')[0].appendChild(ele)
}

function loadGitalk () {
  function initGitalk () {
    var gitalk = new Gitalk(Object.assign({
      clientID: '724b7aa968ee8ea82914',
      clientSecret: '416b14a2294b82ce2040ee2e44c2af9c74ef4851',
      repo: 'JoyceLiang-sudo.github.io',
      owner: 'JoyceLiang-sudo',
      admin: ['JoyceLiang-sudo'],
      id: 'b9bf21dca089256410fa852bc483ef53',
      updateCountCallback: commentCount
    },null))

    gitalk.render('gitalk-container')
  }

  if (typeof Gitalk === 'function') initGitalk()
  else {
    addGitalkSource()
    getScript('https://unpkg.com/gitalk/dist/gitalk.min.js').then(initGitalk)
  }
}

function commentCount(n){
  let isCommentCount = document.querySelector('#post-meta .gitalk-comment-count')
  if (isCommentCount) {
    isCommentCount.innerHTML= n
  }
}

if ('Gitalk' === 'Gitalk' || !false) {
  if (false) btf.loadComment(document.getElementById('gitalk-container'), loadGitalk)
  else loadGitalk()
} else {
  function loadOtherComment () {
    loadGitalk()
  }
}</script></div><script async data-pjax src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div></body></html>